### PR TITLE
Overwatch Console TGUI Refactor

### PR DESCRIPTION
**Pull request by TheGamerdk**
# About the pull request

This redoes the whole console in TGUI.

Adds 2 new features:

Saving coordinates. You can now save up to 3 sets of coordinates, and attach comments. For easy dialling of supply drops and OBs
Multicam, observe up to 4 marines using the console. The window is smaller, and so is the detail. 
This means you can overwatch up to 5 people (1 fullscreen, 4 multicam)

TODO:

- [ ] Investigate multicam shit
- [ ] Sort by role

# Explain why it's good for the game

HTML ugly. TGUI Pretty.
Also gives a few neat CIC buffs

See screenshots below.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

image
image
image
image
image
image
</details>

# Changelog

:cl:
refactor: Refactors the overwatch console in TGUI
/:cl:

### CODE REVIEW

**Comment by TheGamerdk** [comment]
> I don't know how to see the linter errors :(

i figured it out :)

File: code/modules/cm_marines/overwatch.dm [hunk 1]
```diff
@@ -57,437 +76,318 @@
...
-					}
-				} catch(err) {}
-			}
-		}
-	</script>
-	"}
+	data["squad_list"] = list()
+	for(var/datum/squad/S in RoleAuthority.squads)
+		if(S.active && !S.overwatch_officer && S.faction == faction && S.name != "Root")
+			data["squad_list"] += S.name
```
Comment (harryob) [inline code/modules/cm_marines/overwatch.dm:110]:
```suggestion
	for(var/datum/squad/current_squad in RoleAuthority.squads)
		if(current_squad.active && !current_squad.overwatch_officer && current_squad.faction == faction && current_squad.name != "Root")
			data["squad_list"] += current_squad.name
```

File: code/modules/cm_marines/overwatch.dm [hunk 2]
```diff
@@ -57,437 +76,318 @@
...
+	data["spec_alive"] = spec_alive
+	data["medic_alive"] = medic_alive
+	data["engi_alive"] = engi_alive
+	data["smart_alive"] = smart_alive
+	data["specialist_type"] = specialist_type ? specialist_type : "NONE"
+
+	data["z_hidden"] = z_hidden
+
+	data["saved_coordinates"] = list()
+	for(var/i=1, i < length(saved_coordinates) + 1, i++)
```
Comment (harryob) [inline code/modules/cm_marines/overwatch.dm:297]:
```suggestion
	for(var/i in 1 to length(saved_coordinates))
```

File: tgui/packages/tgui/interfaces/OverwatchConsole.js [hunk 3]
```diff
@@ -0,0 +1,820 @@
...
+        {(!data.current_squad && <HomePanel />) || <SquadPanel />}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const HomePanel = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  // Buttons don't seem to support hexcode colors, so we'll have to do this manually, sadly
```
[the remark this answers is not in the document]
Comment (TheGamerdk) [reply tgui/packages/tgui/interfaces/OverwatchConsole.js:20]: Negative, removes hover effect + changes text color

[the remark this answers is not in the document]
Comment (harryob) [reply tgui/packages/tgui/interfaces/OverwatchConsole.js:20]: that's fine 👅

**Comment by SupernovaXTS** [comment]
When a non SL is assigned as aSL they are shown as dead

**Comment by SupernovaXTS** [comment]
Removing multicam does not work

**Comment by TheGamerdk** [comment]
Both issues fixed

**Comment by sg2002** [comment]
There's one feature that's been lost in the rework: ability toggle hide shipside or groundside marines. Was pretty useful to quickly gauge squad state.

Also, I feel like this UI is not as ergonomic as the old one, since it takes way more screen space now, but that's something I can get used to, I guess.

Another minor thing that I don't like is that how helmetless marine names get marked red. This color coding keeps confusing me that they're dead, since the same color used for the death state too.

P.S. Sorry for typing it here, but there seems to be no other feedback thread.

**Comment by morrowwolf** [comment]
> There's one feature that's been lost in the rework: ability toggle hide shipside or groundside marines. Was pretty useful to quickly gauge squad state.
> 
> Also, I feel like this UI is not as ergonomic as the old one, since it takes way more screen space now, but that's something I can get used to, I guess.
> 
> Another minor thing that I don't like is that how helmetless marine names get marked red. This color coding keeps confusing me that they're dead, since the same color used for the death state too.
> 
> P.S. Sorry for typing it here, but there seems to be no other feedback thread.

Constructive and thought out feedback is not going to be thrown out. This is good, don't make a habit out of it though *shakes cane*

In other news, I think multicam is *awesome*. Sadly, I don't think it's going to be able to work due to performance issues. Switching between overwatch consoles quickly is part of CIC gameplay and the load time is a stutter for some and a *gnarly* hang for seconds for others.

Will play with it some more just it's really important to be able to switch between cams rapidly and efficiently while also moving between consoles/squads. This UI must, *at least*, come close to the ease of use as the current UI.

**Comment by TheGamerdk** [comment]
If anyone wants to pick this up:

Make the Squad Orders + RoleTable collapsible elements to reduce the size of the UI
Change the color of (NO HELMET)
Remove the Multicam stuff

Experiment with:
Reducing element size to reduce the width of the UI

**Pull request closed without merging**